tests/random_test.py

```diff
@@ -2039,12 +2056,9 @@ def _double_threefry_seed(seed):
   return jnp.vstack([threefry_seed(s1),
                      threefry_seed(s2)])
 
-def _double_threefry_split(key, num):
-  split0 = threefry_split(key[0], num)
-  split1 = threefry_split(key[1], num)
-  merge = jnp.vstack([jnp.expand_dims(split0, axis=0),
-                      jnp.expand_dims(split1, axis=0)])
-  return merge.transpose((1, 0, 2))
+def _double_threefry_split(key, shape):
+  return vmap(
+      threefry_split, (0, None), -2)(key.reshape(2, 2), shape)
```

Comment:
Change `-2` to `len(shape)` here as well.

Also, can we drop the `reshape`? I think `key` already has shape `(2, 2)`?